src/main/java/com/cs3365/rpgdapi/weapons/WeaponsController.java

```diff
@@ -49,14 +51,17 @@ public WeaponsController(WeaponsService weaponsService) {
                              .body("");
     }
     
-    @RequestMapping(value = "/files/{id}")
-    public ResponseEntity<WeaponEntity> get(@PathVariable("id") UUID identifier)
+    @GetMapping("/{weaponID}")
+    public ResponseEntity get(@PathVariable("weaponID") UUID identifier)
     {
         try {
-            return ResponseEntity.ok(weaponsService.findWeapon(identifier));
+            return new ResponseEntity<>(weaponsService.findWeapon(identifier), HttpStatus.OK);
         }
         catch(WeaponsException e) {
-            return ResponseEntity.badRequest().header("Could not retrieve the weapon from the database").body(null);
+            return ResponseEntity.badRequest()
+                                 .body(String.format(
+                                         "Could not retrieve the weapon from the database",
+                                         e.getMessage()));
```

Comment:
You need to add a `%s` to your string in `String.format()` if you want the message to be returned. For example: `String.format("Could not retrieve the weapon from the database: %s", e.getMessage())`